daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/VariableMap1.scala

```diff
@@ -312,7 +314,7 @@ class VariableMap private(vTable: Map[GlobalQName, ArrayBuffer[VariableInstance]
   def find(qName: GlobalQName): Option[VariableInstance] = {
     val optBuf = vTable.get(qName)
     val variab = {
-      if (optBuf.isDefined)
+      if (optBuf.isDefined && !optBuf.get.isEmpty)
```

Comment:
This feels like it shouldn't be needed and is maybe a sign that we're doing somethign wrong?